src/rules/javascript/complex_prefs_defaults.js

```diff
@@ -0,0 +1,26 @@
+import { COMPLEX_PREFS_DEFAULTS } from 'messages/javascript';
+
+export default function(context) {
+  var filename = context.getFilename();
+
+  if (filename.indexOf('defaults/preferences/') === 0) {
```

Comment:
Come to think of it, I believe the `JavaScriptScanner` class won't fail without a `filename` (or with an `undefined`/`null` one).

We should either demand a filename or we should check for `filename` existing (that'd be a separate issue). Thoughts?